### PR TITLE
DEPR: Enforce deprecations in indexes/datetimes.py

### PR DESCRIPTION
**Pull request by mroeschke**
Introduced in https://github.com/pandas-dev/pandas/pull/43504, https://github.com/pandas-dev/pandas/pull/37819

### CODE REVIEW

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -433,9 +433,11 @@ Removal of prior version deprecations/changes
 - Remove :meth:`DataFrameGroupBy.pad` and :meth:`DataFrameGroupBy.backfill` (:issue:`45076`)
 - Remove ``numpy`` argument from :func:`read_json` (:issue:`30636`)
 - Disallow passing abbreviations for ``orient`` in :meth:`DataFrame.to_dict` (:issue:`32516`)
+- Disallow partial slicing on an unordered :class:`DatetimeIndex` with keys, which are not in Index. This now raises a ``KeyError`` (:issue:`18531`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v2.0.0.rst:436]: "keys, which are not in Index" -> "keys which are not in the Index"?
"unordered" -> "non-monotonic"?

Comment (mroeschke) [reply doc/source/whatsnew/v2.0.0.rst]: Thanks. Updated + greenish

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Comment by jbrockmendel** [comment]
thanks @mroeschke

**Pull request merged**